### PR TITLE
[SPARK-10273] Add @since annotation to pyspark.mllib.feature

### PR DESCRIPTION
**Pull request by noel-smith**
Duplicated the @since decorator from pyspark.sql into pyspark (also tweaked to handle functions without docstrings).

Added @since to methods + "versionadded::" to classes (derived from the git file history in pyspark).

### CODE REVIEW

File: python/pyspark/__init__.py [hunk 1]
```diff
@@ -51,6 +51,26 @@
 # for back compatibility
 from pyspark.sql import SQLContext, HiveContext, SchemaRDD, Row
 
+
+def since(version):
```
Comment (mengxr) [inline python/pyspark/__init__.py]: This should belong to SPARK-10373. We should have a separate PR for it and then update this PR. ping @davies

**Comment by mengxr** [comment]
add to whitelist

File: python/pyspark/__init__.py (hunk 1 above)
Comment (davies) [inline python/pyspark/__init__.py]: https://github.com/apache/spark/pull/8657

Comment (noel-smith) [inline python/pyspark/__init__.py]: OK - no problem - I'll update this (+ SPARK-10269/10271/10272) once SPARK-10373 is complete.

File: python/pyspark/mllib/feature.py [hunk 2]
```diff
@@ -84,11 +84,14 @@ class Normalizer(VectorTransformer):
     >>> nor2 = Normalizer(float("inf"))
     >>> nor2.transform(v)
     DenseVector([0.0, 0.5, 1.0])
+
+    .. versionadded:: 1.2.0
```
Comment (davies) [inline python/pyspark/mllib/feature.py]: Since we won't introduce new API in the micro releases, I think it's better to use `1.2` for short. cc @mengxr

Comment (noel-smith) [inline python/pyspark/mllib/feature.py]: I think matching the overall project versioning scheme make it's clearer - but I'm happy to implement it either way. 

One thing to watch for with using floats is that you can't differentiate between 1.1 and 1.10 (but it looks like that's unlikely to be a problem from the versioning history).

Comment (davies) [inline python/pyspark/mllib/feature.py]: For `1.10` or `1.1.1`, we still need to use string. This change is not necessary, just one minor comment.

**Comment by mengxr** [comment]
@noel-smith @yu-iskw Could you help review each other's PRs on `pyspark.ml` `@since` versions?

* https://spark-prs.appspot.com/users/noel-smith
* https://spark-prs.appspot.com/users/yu-iskw

I want to merge them as soon as possible to avoid future merge conflicts. It would be nice if we can distribute the work. Thanks!

**Comment by noel-smith** [comment]
@mengxr @yu-iskw - Sounds like a plan - I'll take a look.

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks!

**Pull request closed without merging**